src/Microsoft.Identity.Abstractions/ApplicationOptions/RequestContext.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.Identity.Abstractions
+{
+    /// <summary>
+    /// Holds information about an incoming request.
+    /// </summary>
+    public class RequestContext
```

Comment:
not sure why we need this, can you give more details?

Reply:
The PR description provides some detail on the usefulness of the RequestContext.

It is not needed today but offers flexibility for the future. The PropertyBag on the RequestContext can hold additional information about the request from the user or Downstream API, preventing the need for a new interface. This is similar to Wilson's CallContext, which is used throughout the call graph.

Reply:
Removed RequestContext as it is not needed today.